test/boost/memtable_test.cc

```diff
@@ -100,6 +100,12 @@ SEASTAR_TEST_CASE(test_memtable_with_many_versions_conforms_to_mutation_source)
         };
         auto cleanup_readers = defer([&] { clear_readers(); });
         std::deque<dht::partition_range> ranges_storage;
+        lw_shared_ptr<bool> finished = make_lw_shared(false);
```

Comment:
Re commit message:

> ... as the tests always managed to read all range
tombstones and rows, which cover a given clustering range.

So now the tests don't manage to read all range tombstones and rows?